src/scripts/main.js

```diff
@@ -1,3 +1,32 @@
 'use strict';
 
-// write code here
+function convertToNumber(stringSalary) {
+  return parseFloat(stringSalary.replace(/[^\d.-]/g, ''));
+}
+
+function sortList(list) {
+  const itemsList = [...list.children];
+
+  itemsList.sort((a, b) => {
+    const salaryA = convertToNumber(a.dataset.salary);
+    const salaryB = convertToNumber(b.dataset.salary);
+
+    return salaryB - salaryA;
+  });
+
+  list.innerHTML = '';
+
+  itemsList.forEach((item) => list.appendChild(item));
+}
+
+function getEmployees(list) {
+  const items = [...list.children];
+
+  return items.map((i) => ({
+    name: i.dataset.name,
+    position: i.dataset.position,
+    salary: convertToNumber(i.dataset.salary),
+  }));
+}
+sortList(document.querySelectorAll('li'));
```

Comment:
The `sortList` function is being called with `document.querySelectorAll('li')`, which returns a NodeList. This function expects a single list element, not a NodeList. You should pass a single list element to this function.